helios-services/src/main/java/com/spotify/helios/servicescommon/coordination/DefaultZooKeeperClient.java

```diff
@@ -389,6 +395,51 @@ public Stat stat(final String path) throws KeeperException {
     return transaction(asList(operations));
   }
 
+  @Override
+  public void setAcl(final String path, final List<ACL> aclList) throws KeeperException {
+    assertClusterIdFlagTrue();
+    try {
+      client.setACL().withACL(aclList).forPath(path);
+    } catch (Exception e) {
+      propagateIfInstanceOf(e, KeeperException.class);
+      throw propagate(e);
+    }
+  }
+
+  @Override
+  public List<ACL> getAcl(final String path) throws KeeperException {
+    assertClusterIdFlagTrue();
+    try {
+      return client.getACL().forPath(path);
+    } catch (Exception e) {
+      propagateIfInstanceOf(e, KeeperException.class);
+      throw propagate(e);
+    }
+  }
+
+  @Override
+  public void initializeAclRecursive(final String path, final ACLProvider aclProvider)
+      throws KeeperException {
+    assertClusterIdFlagTrue();
+    try {
+      final List<ACL> expected = aclProvider.getAclForPath(path);
+      final List<ACL> actual = getAcl(path);
+
+      if ((expected.size() == actual.size()) && (expected.containsAll(actual))) {
```

Comment:
elsewhere in this diff two `List<ACL>`s are compared by converting both to sets and calling `.equals()`, should do the same here for consistency